tests/testthat/test-carehomes-multistrain.R

```diff
@@ -764,3 +760,248 @@ test_that("calculate Rt with both second variant and vaccination", {
+                            strain_rel_gamma_A = c(1, 1.5),
+                            strain_rel_gamma_P = c(1, 1.5),
+                            strain_rel_gamma_C_1 = c(1, 1.5),
+                            strain_rel_gamma_C_2 = c(1, 1.5),
+                            strain_seed_date =
+                              rep(sircovid_date("2020-02-07"), 2),
+                            strain_seed_rate = c(10, 0))
+  p$gamma_A[1] <- p$gamma_A[1] * 1.5
+  p$gamma_P[1] <- p$gamma_P[1] * 1.5
+  p$gamma_C_1[1] <- p$gamma_C_1[1] * 1.5
+  p$gamma_C_2[1] <- p$gamma_C_2[1] * 1.5
+
+  np <- 3L
+  mod <- carehomes$new(p, 0, np, seed = 1L)
+
+  initial <- carehomes_initial(mod$info(), 10, p)
+  mod$set_state(initial$state, initial$step)
+  index_S <- mod$info()$index$S
+  index_prob_strain <- mod$info()$index$prob_strain
+
+  end <- sircovid_date("2020-05-01") / p$dt
+  steps <- seq(initial$step, end, by = 1 / p$dt)
+
+  set.seed(1)
+  y <- mod$simulate(steps)
+  S <- y[index_S, , ]
+  prob_strain <- y[index_prob_strain, , ]
+
+  rt_15 <- carehomes_Rt(steps, S[, 1, ], p, prob_strain[, 1, ])
+  rt_15_all <- carehomes_Rt_trajectories(steps, S, p, prob_strain)
+
+  ## rate equal to ref
+  p <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
+                            strain_transmission = c(1, 1),
+                            strain_seed_date =
+                              rep(sircovid_date("2020-02-07"), 2),
+                            strain_seed_rate = c(10, 0))
+
+  np <- 3L
+  mod <- carehomes$new(p, 0, np, seed = 1L)
+
+  initial <- carehomes_initial(mod$info(), 10, p)
+  mod$set_state(initial$state, initial$step)
+  index_S <- mod$info()$index$S
+  index_prob_strain <- mod$info()$index$prob_strain
+
+  end <- sircovid_date("2020-05-01") / p$dt
+  steps <- seq(initial$step, end, by = 1 / p$dt)
+
+  set.seed(1)
+  y <- mod$simulate(steps)
+  S <- y[index_S, , ]
+  prob_strain <- y[index_prob_strain, , ]
+
+  rt_1 <- carehomes_Rt(steps, S[, 1, ], p, prob_strain[, 1, ])
+  rt_1_all <- carehomes_Rt_trajectories(steps, S, p, prob_strain)
+
+  ## Rt should be higher (or equal) for the two variant version
+  expect_true(all(rt_1$Rt_all > rt_15$Rt_all))
+  expect_true(all(rt_1$Rt_general > rt_15$Rt_general))
+  expect_true(all(rt_1_all$Rt_all > rt_15$Rt_all))
+  expect_true(all(rt_1_all$Rt_general > rt_15$Rt_general))
+})
+
+
+test_that("Stuck when gamma =  0", {
+  np <- 3L
+
+  ## gammaP is 0 so IC1 is 0
+  p <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
+                            strain_transmission = c(1, 1),
+                            strain_rel_gamma_A = 1,
+                            strain_rel_gamma_P = 1,
+                            strain_rel_gamma_C_1 = 1,
+                            strain_rel_gamma_C_2 = 1,
+                            strain_seed_date =
+                              rep(sircovid_date("2020-02-07"), 2),
+                            strain_seed_rate = c(10, 0))
+  p$gamma_P[] <- 0
```

Comment:
I would maybe add a test where you set strain_rel_gamma_P <- c(1, 0) and then check that you still have progression for strain 1 but not for strain 2